### PR TITLE
Refactor secure baseline documents to be within scubagear module

### LINKED ISSUE #679: Refactor Baseline documents to be within ScubaGear module
# 💡 Summary #

ScubaGear is currently deployed as a zipped package with manual instructions on installation and usage. To migrate to delivery/deployment via a package repository all baseline documents should be included in publishing.

## Motivation and context ##

This would be useful because it supports publishing ScubaGear to a public repository #645 

## Implementation notes ##

Reference #94, #645, #677 for context and additional information 
-ScubaGear functinality has access to packaged baseline documents (e.g., Import-SecureBaseline)
-Allow users to export any secure baseline

## Acceptance criteria ##

How do we know when this work is done?

- [ ] ScubaGear executes properly when installed from a package repository
- [ ]  ScubaGear executes properly when imported directly from source code baseline
- [ ] User can export any/all baseline documents from ScubaGear when installed froma repository

### PR DESCRIPTION
**Pull request by crutchfield**
#  #

## 🗣 Description ##

Move baselibe documents to be included with module.  Provided a function to copy to a user specified folder - default to user home (e.g., <home>\ScubaGear)
Also cleaned up related files with linter issues. 

## 💭 Motivation and context ##

Closes #679 
Required to package ScubaGear as aPowerShell module deployed from PS Gallery. 

## 🧪 Testing ##

All unit test pass
Generated reports for all products

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [X] Tests have been added and/or modified to cover the changes in this PR. Linter issue #803 
- [X] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by Sloane4 — APPROVED** [review]
Runs against tenant w/o errors

**Comment by crutchfield** [comment]
@isab-m Is your workload going to allow you to review this in the near future.  I can request another reviewer if not.

**Comment by adhilto** [comment]
One word of caution: some public sources have been linking directly to the https://github.com/cisagov/ScubaGear/tree/main/baselines folder. If we move it, those links will break, possibly damaging the visibility of the project. For example:
- https://practical365.com/revised-security-baselines-for-microsoft-365-and-intune/
- https://portal.cci-entel.cl/Threat_Intelligence/Boletines/1808/

I just spent a bit of time digging around and almost all of the articles I found just linked to the Github page (not the baselines folder), so we might still be ok to make this change. But it's a factor we should consider.

**Comment by crutchfield** [comment]
> Running a report shows that the rego utility function PolicyLink still points to the old baseline location and needs to be updated in Rego\Utils\ReportDetails.reog lines 21-31 so it generates the correct URL for the new baseline locations. Noticed this when looking over HTML report results in AAD.

I should add a check in Smoke Test to see if all links are valid.

**Comment by Sloane4** [comment]
> Running a report shows that the rego utility function PolicyLink still points to the old baseline location and needs to be updated in Rego\Utils\ReportDetails.reog lines 21-31 so it generates the correct URL for the new baseline locations. Noticed this when looking over HTML report results in AAD.

This is addressed in my PR #827

**Comment by crutchfield** [comment]
> > > Running a report shows that the rego utility function PolicyLink still points to the old baseline location and needs to be updated in Rego\Utils\ReportDetails.reog lines 21-31 so it generates the correct URL for the new baseline locations. Noticed this when looking over HTML report results in AAD.
> > 
> > 
> > This is addressed in my PR #827
> 
> I checked your PR and this issue is not addressed in #827. That PR addresses the lack of a v in the version tag, but not the wholesale shift of the baselines from the root folder into the PowerShell modules folder tree which this PR moves the baselines to. So yes, both PRs adjust this, but 827 addresses an old bug while this PR introduces a new one that requires an additional fix to the PolicyLink construction if I'm reading things correctly.

Corrected the path.  Link will still be bad since the leading v is in  another PR.

File: PowerShell/ScubaGear/Modules/CreateReport/CreateReport.psm1
[the remark this answers is not in the document]
Comment (crutchfield) [reply PowerShell/ScubaGear/Modules/CreateReport/CreateReport.psm1:1]: Fixed Policy Group Anchor link

image

File: PowerShell/ScubaGear/Modules/Support/Support.psm1 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        $SourceFileName = Join-Path -Path $PSScriptRoot -ChildPath "..\..\baselines\$_.md"
+        $TargetFileName = Join-Path -Path $Destination -ChildPath "$_.md"
+        Copy-Item -Path $SourceFileName -Destination $Destination -Force:$Force -ErrorAction Stop  2> $null
+        Set-ItemProperty -Path $TargetFileName -Name IsReadOnly -Value $true
+    }
+}
+
+Export-ModuleMember -Function @(
+    'Copy-ScubaBaselineDocument'
+)
```
[the remark this answers is not in the document]
Comment (crutchfield) [reply PowerShell/ScubaGear/Modules/Support/Support.psm1]: Added newline at EOF

File: README.md [hunk 2]
```diff
@@ -4,36 +4,47 @@
...
-- [Troubleshooting](#troubleshooting)
-  - [Executing against multiple tenants](#executing-against-multiple-tenants)
-  - [Errors connecting to Defender](#errors-connecting-to-defender)
-  - [Exchange Online maximum connections error](#exchange-online-maximum-connections-error)
-  - [Power Platform Errors](#power-platform-errors)
-  - [Microsoft Graph Errors](#microsoft-graph-errors)
-  - [Running the Tool Behind Some Proxies](#running-the-tool-behind-some-proxies)
-  - [Utility Scripts](#utility-scripts)
-- [Project License](#project-license)
+- [ScubaGear](#scubagear)
```
[the remark this answers is not in the document]
Comment (crutchfield) [reply README.md]: I did a "git checkout origin/main -- README.md" thinking it would revert back to original.  But I think main had already changed on me.

**Comment by crutchfield** [comment]
> The latest updates fix all the linking issues in the reports. But I do agree with @buidav that the baselines folder should remain at the root with a README.md in it that states something like:
> 
> The baselines have been moved into the ScubaGear powershell module for easier distribution. For reference to the baselines, please go here:
> 
> That way viewers visiting the GitHub can still easily find the baselines when referencing the site manually without having to dig for the correct directory.

Added readme to top level baselines folder to reference latest tagged version of documents.

File: baselines/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+The baselines have been moved into the ScubaGear powershell module for easier distribution. For reference to the baselines, please go [here](https://github.com/cisagov/ScubaGear/tree/v1.0.0/PowerShell/ScubaGear/baselines)
```
[the remark this answers is not in the document]
Comment (crutchfield) [reply baselines/README.md]: I did not see your suggestion and already listed individually.  Almost the same.

**Comment by crutchfield** [comment]
@nanda-katikaneni  Ready for merge

**Pull request merged**